internal/controller/supervisorconfig/oidcupstreamwatcher/oidc_upstream_watcher.go

```diff
@@ -302,51 +326,82 @@ func (c *oidcWatcherController) validateSecret(upstream *idpv1alpha1.OIDCIdentit
 	return &metav1.Condition{
 		Type:    typeClientCredentialsSecretValid,
 		Status:  metav1.ConditionTrue,
-		Reason:  upstreamwatchers.ReasonSuccess,
+		Reason:  conditionsutil.ReasonSuccess,
 		Message: "loaded client credentials",
 	}
 }
 
 // validateIssuer validates the .spec.issuer field, performs OIDC discovery, and returns the appropriate OIDCDiscoverySucceeded condition.
-func (c *oidcWatcherController) validateIssuer(ctx context.Context, upstream *idpv1alpha1.OIDCIdentityProvider, result *upstreamoidc.ProviderConfig) *metav1.Condition {
-	// Get the provider and HTTP Client from cache if possible.
-	discoveredProvider, httpClient := c.validatorCache.getProvider(&upstream.Spec)
-
-	// If the provider does not exist in the cache, do a fresh discovery lookup and save to the cache.
-	if discoveredProvider == nil {
-		var err error
-		httpClient, err = getClient(upstream)
-		if err != nil {
-			return &metav1.Condition{
+func (c *oidcWatcherController) validateIssuer(ctx context.Context, upstream *idpv1alpha1.OIDCIdentityProvider, result *upstreamoidc.ProviderConfig) []*metav1.Condition {
+	tlsCondition, caBundle := tlsconfigutil.ValidateTLSConfig(
+		tlsconfigutil.TLSSpecForSupervisor(upstream.Spec.TLS),
+		"spec.tls",
+		upstream.Namespace,
+		c.secretInformer,
+		c.configMapInformer)
+
+	// When the TLS config is invalid, return some error conditions.
+	if tlsCondition.Reason != conditionsutil.ReasonSuccess {
```

Comment:
Should this actually create a "unknown" condition and also add the `tlsCondition` to the status conditions? This implementation seems inconsistent with the other controllers.

Reply:
It would be nice if this created an "Unknown" condition, but the current behavior is how it has always worked, so we decided to address it in a future PR.

When the tlsCondition is happy, it does appear as a new condition on the resource.